### PR TITLE
Appknox-Go: Region and Host functionality

### PR DESCRIPTION
**Pull request by yashviagrawal**
Added a region flag, and now host takes url only, and region takes names only

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ Flags:
   -k, --insecure              Disable Security Checks
       --pac string            pac file path or url
       --proxy string          proxy url
+      --region string         Region names like global, saudi, uae
```
Comment (ginilpg) [inline README.md]: @yashviagrawal "Region names, e.g., global, saudi, uae. By default, global is used"

File: helper/clientInitialize.go [hunk 2]
```diff
@@ -24,30 +24,77 @@ func getAppknoxAccessToken() string {
...
+            availableRegions := make([]string, 0, len(hostMappings))
+            for key := range hostMappings {
+                availableRegions = append(availableRegions, key)
+            }
+            fmt.Printf("Invalid region name: %s. Available regions: %s\n", region, strings.Join(availableRegions, ", "))
+            os.Exit(1)
+        }
+    } else {
+        // Validate the host is a proper URL if no region is passed
+        _, err := url.ParseRequestURI(host)
```
Comment (ginilpg) [inline helper/clientInitialize.go]: @yashviagrawal If both region and host is provided, host will take the priority over region. How. are we handling that overriding scenraio here?

**Review by ginilpg — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (yashviagrawal) [reply README.md]: Yes, added this.

File: helper/clientInitialize.go (hunk 2 above)
Comment (yashviagrawal) [reply helper/clientInitialize.go]: Basically, in root.go default api host was set, removed that and now the precendence logic works

File: cmd/root.go [hunk 3]
```diff
@@ -35,10 +35,15 @@ func init() {
...
-	viper.SetDefault("host", appknox.DefaultAPIHost)
+	RootCmd.PersistentFlags().String("host", "", "Appknox Server") // No default value here
+    viper.BindPFlag("host", RootCmd.PersistentFlags().Lookup("host"))
+    viper.BindEnv("host", "APPKNOX_API_HOST")
+
+
+	RootCmd.PersistentFlags().String("region", "", "Region names, e.g., global, saudi, uae. By default, global is used")
+    viper.BindPFlag("region", RootCmd.PersistentFlags().Lookup("region"))
+    viper.BindEnv("region", "APPKNOX_API_REGION")
+    viper.SetDefault("region", "")
```
Comment (cosmosgenius) [inline cmd/root.go]: The default is suppose to be `global`. Can we add it here?

File: helper/clientInitialize.go [hunk 4]
```diff
@@ -24,30 +24,86 @@ func getAppknoxAccessToken() string {
...
+                }
+                fmt.Printf("Invalid region name: %s. Available regions: %s\n", region, strings.Join(availableRegions, ", "))
+                os.Exit(1)
+            }
+        } else {
+            // If neither host nor region are provided, default to the global host
+            host = hostMappings["global"]
+        }
+    } else {
+        // If both region and host are provided, prioritize host and ignore region
```
Comment (cosmosgenius) [inline helper/clientInitialize.go]: can we move this logic to a separate function along with tests? There is too much nested if here without tests?

the function should probably accept the `host` env and `region` env and return (baseurl, error) with similar logic.

File: helper/clientInitialize.go [hunk 5]
```diff
@@ -24,30 +24,86 @@ func getAppknoxAccessToken() string {
...
+         "saudi":   "https://sa.secure.appknox.com/",
+         // Add more mappings as needed
+     }
+ }
+
+func getAPIHostMappings() map[string]string {
+    // Instead of using an environment variable, call the new function
+    return GetHostMappings()
+}
+
```
Comment (cosmosgenius) [inline helper/clientInitialize.go]: Is the function needed? `GetHostMappings` doing the same thing.

**Review by cosmosgenius — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 3 above)
Comment (yashviagrawal) [reply cmd/root.go]: Yes, added.

File: helper/clientInitialize.go (hunk 5 above)
Comment (yashviagrawal) [reply helper/clientInitialize.go]: Yes, the function is redundant, fixed that.

File: helper/clientInitialize.go (hunk 4 above)
Comment (yashviagrawal) [reply helper/clientInitialize.go]: Yes, moved the logic to a separate function.

**Review by cosmosgenius — APPROVED** [review]

**Pull request merged**